tb_plugin/torch_tb_profiler/profiler/module_parser.py

```diff
@@ -355,34 +184,15 @@ def parse_kernels(kernel_list):
             kernel_list_groupby_name_op = list(name_op_to_agg.values())
             return kernel_list_groupby_name_op
 
-        # For OperatorNode and ProfilerStepNode:
-        #   Use time interval containing relationship to build father-child correlation,
-        #   which is consistent with autograd profiler.
-        # For RuntimeNode:
-        #   Use external_id to build correlation with its father OperatorNode or ProfilerStepNode.
-        #   Because in the case when RuntimeNode has duration 0 and starts at same time as a OperatorNode,
-        #   just use interval containing relationship can't tell it is child or brother of the OperatorNode.
-        tid2list = {}  # value is a list of OperatorNode and ProfilerStepNode. Do not include RuntimeNode
-        tid2zero_rt_list = {}  # value is a list of RuntimeNode with external_id=0. They will be attached to root nodes.
-        corrid_to_device = {}  # value is a list of DeviceNode
-        corrid_to_runtime = {}  # value is a RuntimeNode
-        externalid_to_runtime = {}  # value is a list of RuntimeNode
-        for event in events:
-            parse_event(event, corrid_to_device, corrid_to_runtime, externalid_to_runtime, tid2list, tid2zero_rt_list)
+        tid2list = context.tid2list
+        tid2zero_rt_list = context.tid2zero_rt_list
+        corrid_to_device = context.corrid_to_device
+
         # Kernel that not owned by any operator should also be shown in kernel view
         # when group by "Kernel Name + Op Name".
         for _, device_nodes in corrid_to_device.items():
             self.kernel_list.extend([n for n in device_nodes if n.type == EventTypes.KERNEL])
```

Comment:
If this and line116 can merge to node_parser.py:line58, then the corrid_to_device can also change to local variable.